RenderStatePackagerLib/interface/DeviceObjectDescriptorParser.h

```diff
@@ -0,0 +1,305 @@
+
+    Uint32                                        ProceduralHitShadersCount DEFAULT_INITIALIZER(0);
+
+#if DILIGENT_CPP_INTERFACE
+    constexpr bool operator == (const RayTracingPipelineStateCreateInfoJSON& RHS) const 
+    {
+        //TODO
+        return false;
+    }
+#endif
+};
+typedef struct RayTracingProceduralHitShaderGroupJSON RayTracingProceduralHitShaderGroupJSON;
+
+// {355AC9f7-5D9D-423D-AE35-80E0028DE17E}
+static const INTERFACE_ID IID_DeviceObjectDescriptorParser = {0x355AC9F7, 0x5D9D, 0x423D, {0xAE, 0x35, 0x80, 0xE0, 0x02, 0x8D, 0xE1, 0x7E}};
+
+#define DILIGENT_INTERFACE_NAME IDeviceObjectDescriptorParser
+#include "../../../DiligentCore/Primitives/interface/DefineInterfaceHelperMacros.h"
+
+#define DeviceObjectDescriptionParserInclusiveMethods \
+    IObjectInclusiveMethods;                          \
+    IDeviceObjectDescriptorParser DeviceObjectDescriptorParser
+
+
+DILIGENT_BEGIN_INTERFACE(IDeviceObjectDescriptorParser, IObject)
+{
+    VIRTUAL Bool METHOD(GetGraphicsPipelineState)(THIS_
+                                                  const Char* Name,
+                                                  GraphicsPipelineStateCreateInfo REF CreateInfo,
+                                                  GraphicsPipelineStateCreateInfoJSON REF CreateInfoJSON) CONST PURE;
+
+    VIRTUAL Bool METHOD(GetComputePipelineState)(THIS_
+                                                 const Char* Name,
+                                                 ComputePipelineStateCreateInfo REF CreateInfo,
+                                                 ComputePipelineStateCreateInfoJSON REF CreateInfoJSON) CONST PURE;
+
+    VIRTUAL Bool METHOD(GetRayTracingPipelineState)(THIS_
+                                                    const Char* Name,
+                                                    RayTracingPipelineStateCreateInfo REF CreateInfo,
+                                                    RayTracingPipelineStateCreateInfoJSON REF CreateInfoJSON) CONST PURE;
+
+    VIRTUAL Bool METHOD(GetTilePipelineState)(THIS_
+                                              const Char* Name,
+                                              TilePipelineStateCreateInfo REF CreateInfo,
+                                              TilePipelineStateCreateInfoJSON REF CreateInfoJSON) CONST PURE;
+
+    VIRTUAL Bool METHOD(GetResourceSignature)(THIS_
+                                              const Char* Name,
+                                              PipelineResourceSignatureDesc REF CreateInfo) CONST PURE;
+
+    VIRTUAL Bool METHOD(GetShader)(THIS_
+                                   const Char* Name,
+                                   ShaderCreateInfo REF CreateInfo) CONST PURE;
+
+    VIRTUAL Bool METHOD(GetRenderPass)(THIS_
+                                       const Char* Name,
+                                       RenderPassDesc REF CreateInfo) CONST PURE;
+
+    
+    VIRTUAL Bool METHOD(EnumGraphicsPipelineStates)(THIS_
```

Comment:
This is not a very clear API. I suggest adding GetPipelineStateCount method() and
GetPipelineStateByIndex().
GetPipelineState can be renamed to GetPipelineStateByName to be consistent with shader reflection API